util/proto.go

```diff
@@ -0,0 +1,79 @@
+package util
```

Comment:
This shouldn't be generic util pkg. Maybe `util/cast` ?

Reply:
Sounds good.

Reply:
Addressed by fc4c2fe3c879d084f42639c5dd6b28d0e9f87c48